### PR TITLE
added dsl and tests and cleaned up codenarc violations

### PR DESCRIPTION
**Pull request by estute**
@aphanse @jzoldak spock tests and sandbox tests passed. Running on test jenkins. Take a peek when you get a chance.

### CODE REVIEW

File: platform/jobs/edxPlatformQualityPr.groovy [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        wrappers {
+            timeout {
+                absolute(45)
+            }
+            timestamps()
+            colorizeOutput()
+            buildName('#\${BUILD_NUMBER}: Quality Tests')
+        }
+        steps {
+            shell('cd edx-platform; TEST_SUITE=quality ./scripts/all-tests.sh')
```
Comment (aphanse) [inline platform/jobs/edxPlatformQualityPr.groovy]: change from edx-platform to jobConfig['repoName'] to work with the other repo :)

File: platform/jobs/edxPlatformQualityPr.groovy [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        }
+        publishers {
+            archiveArtifacts {
+                pattern('edx-platform/reports/**/*,edx-platform/test_root/log/*.png,edx-platform/test_root/log/*.log,edx-platform/test_root/log/hars/*.har,edx-platform/**/nosetests.xml,edx-platform/**/TEST-*.xml')
```
Comment (aphanse) [inline platform/jobs/edxPlatformQualityPr.groovy]: likewise use edx-platform* to ensure everything works :)

**Comment by aphanse** [comment]
👍 really enjoyed all of the fruit

**Comment by estute** [comment]
@aphanse just because we have had so much back and forth on this one, can you check it out again? thx

File: build.gradle [hunk 3]
```diff
@@ -63,13 +63,15 @@ dependencies {
     testCompile "org.jenkins-ci.plugins:job-dsl:${jobDslVersion}@jar"
     testCompile 'org.jenkins-ci.plugins:structs:1.1@jar'
     testCompile 'org.jenkins-ci.plugins:cloudbees-folder:4.4@jar'
+    testCompile 'com.cloudbees.plugins:build-flow-plugin:0.17@jar'
+    //testCompile 'org.jenkins-ci.plugins:buildgraph-view:1.3.2@jar'
```
Comment (jzoldak) [inline build.gradle]: Should this be removed?

File: platform/jobs/edxPlatformQualityPr.groovy [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    repoName : name-of-github-edx-repo
+    platformUrl : platform-github-url-segment.git
+    platformCredential : n/a
+    platformCloneReference : clone/.git
+    protocol : https
+*/
+
+/* stdout logger */
+/* use this instead of println, because you can pass it into closures or other scripts. */
+/* TODO: Move this into JenkinsPublicConstants, as it can be shared. */
```
Comment (jzoldak) [inline platform/jobs/edxPlatformQualityPr.groovy]: We should either take the time to refactor this, or stop marking it as a TODO in all the new files. Do you know if there is a ticket in the backlog that we can estimate and discuss at sprint planning?

File: platform/jobs/edxPlatformQualityPr.groovy [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    return 1
+}
+
+/* Iterate over the job configurations */
+secretMap.each { jobConfigs ->
+
+    Map jobConfig = jobConfigs.getValue()
+
+    /* Test secret contains all necessary keys for this job */
+    assert jobConfig.containsKey('open')
```
Comment (aphanse) [inline platform/jobs/edxPlatformQualityPr.groovy]: What happened to the rest of the assertions/why is this the only one we assert on?

File: platform/jobs/edxPlatformQualityPr.groovy (hunk 4 above)
Comment (estute) [inline platform/jobs/edxPlatformQualityPr.groovy]: Writing a ticket now.

File: platform/jobs/edxPlatformQualityPr.groovy (hunk 5 above)
Comment (estute) [inline platform/jobs/edxPlatformQualityPr.groovy]: I took them out while getting this one running. I don't love these assertions, but you are right, we should keep them.

File: build.gradle (hunk 3 above)
Comment (estute) [inline build.gradle]: Yes.

File: platform/jobs/edxPlatformQualityPr.groovy (hunk 5 above)
Comment (jzoldak) [inline platform/jobs/edxPlatformQualityPr.groovy]: Our other job definitions assert on all required keys. We should do that here also, especially as we don't have error handling in the rest of the file.

**Comment by estute** [comment]
@aphanse @jzoldak

**Comment by aphanse** [comment]
👍

**Pull request merged**